app/javascript/src/Navigation/components/VerticalNav.jsx

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+import React from 'react'
+import { Nav, NavExpandable, NavItem, NavList, NavGroup } from 'Navigation/components/PF4NavProxy'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+type Item = {
+  id: string,
+  title: string,
+  path: string,
+  target: ?string
+}
+
+type Section = {
+  id: string,
+  title: string,
+  path: string,
+  items: Item[]
+}
+
+type Props = {
+  sections: Section[],
+  activeSection: ?string,
+  activeItem: ?string
+}
+
+const VerticalNav = ({ sections, activeSection, activeItem }: Props) => (
+  <div className="pf-c-page__sidebar-body">
+    <Nav id='mainmenu'>
+      <NavList>
+        {sections.map(({ id, title, path, items }) => {
+          const sId = id
+          return items
+            ? (
+              <NavExpandable title={title} isActive={activeSection === sId} isExpanded={activeSection === sId}>
```

Comment:
to avoid so much nesting, might be good to extract this to a different component

Reply:
Good idea